### PR TITLE
Use SIP filename for AIP filename

### LINKED ISSUE #656: Problem: AIP names are not named after the transfer name
**Is your feature request related to a problem? Please describe.**

All the AIPs created by enduro have the name "enduro-\<UUID\>.zip", and when uncompressed, they have a folder named enduro-\<UUID\>

**Describe the solution you'd like**

Add a configuration setting that allows for AIPs  to be created using the transfer name, without the ".zip" suffix, so the folder is named <transfer_name>-UUID, and the AIP <transfer_name>-UUID.zip

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Diogenesoftoronto**
Closes #656

### CODE REVIEW

File: internal/workflow/processing.go [hunk 1]
```diff
@@ -123,6 +125,11 @@ type TransferInfo struct {
 	DefaultPermanentLocationID *uuid.UUID
 }
 
+func (t *TransferInfo) Name() string {
+	s := strings.Split(filepath.Base(t.Key), ".")
+	return s[0]
+}
```
Comment (sevein) [inline internal/workflow/processing.go]: Here's an alternative that seems to be faster and doesn't allocate the slice:

```go
func (t *TransferInfo) Name() string {
	base := filepath.Base(t.Key)
	if i := strings.Index(base, "."); i != -1 {
		return base[:i]
	}
	return base
}
```

https://gist.github.com/sevein/f071b4205170dd28482b085778d8ecff

```
$ go test -bench=. -benchmem .
goos: linux
goarch: amd64
pkg: foobar
cpu: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
BenchmarkV1-6   	12694426	        85.29 ns/op	      48 B/op	       1 allocs/op
BenchmarkV2-6   	81417373	        12.79 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	foobar	2.239s
```

File: internal/storage/download.go [hunk 2]
```diff
@@ -47,7 +51,7 @@ func Download(svc Service, mux goahttp.Muxer, dec func(r *http.Request) goahttp.
 		}
 		defer reader.Close()
 
-		filename := fmt.Sprintf("enduro-%s.7z", pkg.AipID)
+		filename := fmt.Sprintf("%s-%s.7z", strings.Split(filepath.Base(pkg.Name), ".")[0], pkg.AipID)
```
Comment (sevein) [inline internal/storage/download.go]: strings.Split(filepath.Base(pkg.Name), ".")[0]

Could you create a `fsutil.FilenameWithoutExt` function in the fsutil package?

**Review by sevein — CHANGES_REQUESTED** [review]
I've suggested a couple of changes.

File: internal/workflow/processing.go (hunk 1 above)
Comment (Diogenesoftoronto) [reply internal/workflow/processing.go]: great test bench example

File: internal/storage/download.go (hunk 2 above)
Comment (Diogenesoftoronto) [reply internal/storage/download.go]: yes, I will take your edited function and use that in fsutil

Comment (Diogenesoftoronto) [reply internal/storage/download.go]: I modified the function a bit. I checked the definition of index and it just uses the indexbyte function if the string is one in len. so i decided to just do that.

File: internal/fsutil/fsutil.go [hunk 3]
```diff
@@ -3,13 +3,23 @@ package fsutil
...
 
 	"github.com/otiai10/copy"
 )
 
 // Used for testing.
 var Renamer = os.Rename
 
+func BaseNoExt(path string) string {
+	base := filepath.Base(path)
+	if idx := strings.IndexByte(base, "."[0]); idx != -1 {
```
Comment (djjuhasz) [inline internal/fsutil/fsutil.go]: I think it's cleaner to use single quotes to return the full stop character as a rune (e.g. '.'). Passing a rune to  `IndexByte` seems to work fine:

```
if idx := strings.IndexByte(base, '.'); idx != -1 {
```

**Review by djjuhasz — CHANGES_REQUESTED** [review]
One minor change request @Diogenesoftoronto.

**Review by djjuhasz — APPROVED** [review]
LGTM :+1:

**Pull request closed without merging**